gobblin-utility/src/main/java/org/apache/gobblin/util/logs/LogCopier.java

```diff
@@ -178,53 +188,91 @@ protected Scheduler scheduler() {
     return Scheduler.newFixedRateSchedule(0, this.sourceLogFileMonitorInterval, this.timeUnit);
   }
 
+  private boolean shouldIncludeLogFile(FileStatus logFile) {
```

Comment:
What happen if one thread pass the branch `if (copiedFileNames.contains(logFilePath.getName())) {`, and the other thread just add a file in `copiedFileNames `?

Reply:
This should be fine, since only the LogCopier thread will prune copiedFileNames while LogCopyTask will add copiedFileNames. The main LogCopier thread blocks until LogCopyTasks finish and then deletes files from the copiedFileNames.